demos/mqtt/mqtt_demo_basic_tls/demo_config.h

```diff
@@ -56,26 +56,30 @@
  * the instructions in https://mosquitto.org/ for running a Mosquitto broker
  * locally.
  */
-#define BROKER_ENDPOINT             "test.mosquitto.org"
+#ifndef BROKER_ENDPOINT
```

Comment:
The philosophy of this file demo_config.h already is that applications can edit it.
In mqtt_demo_basic_tls.c we should have:
```
#ifndef BROKER_ENDPOINT
    #error "Please define an MQTT broker endpoint."
#endif
```

Reply:
I agree, that our demo source files should have build-time failures for no definitions of these macros. Have added them in source files.

Reply:
So how would this build-time failures for no definitions of these macros work for our demo sources? Wont this file always define ```BROKER_ENDPOINT``` ? So by the time we get to our demo source files, wont the check for ```ifndef BROKER_ENDPOINT``` always say that ```BROKER_ENDPOINT``` is defined?

Reply:
It is to guard against accidental removals of the `BROKER_ENDPOINT` by the customer. It provides a helpful message to remind them that the `BROKER_ENDPOINT` macro is required for the demo to run instead of giving them regular gcc build failures (at the source code points that use the macro)

Reply:
This is still different from what the README says about Mutual Auth Demos.